src/main/jjtree/net/sf/jsqlparser/parser/JSqlParserCC.jjt

```diff
@@ -2268,13 +2269,16 @@ PlainSelect PlainSelect() #PlainSelect:
 	[ LOOKAHEAD(<K_LIMIT>, { limit==null }) limit = LimitWithOffset() { plainSelect.setLimit(limit);    } ]
     [ LOOKAHEAD(<K_FETCH>) fetch = Fetch() { plainSelect.setFetch(fetch);    } ]
     [ LOOKAHEAD(<K_WITH> <K_ISOLATION>) withIsolation = WithIsolation() { plainSelect.setIsolation(withIsolation); } ]
-    [ LOOKAHEAD(2) <K_FOR> <K_UPDATE> { plainSelect.setForUpdate(true); }
+    [ LOOKAHEAD(2) <K_FOR> (
+            <K_UPDATE> { plainSelect.setForUpdate(true); } | <K_SHARE> { plainSelect.setForShare(true); } |
```

Comment:
Sorry to ask: are those types not Exlusive like:
- either `FOR UPDATE`
- or `FOR SHARE`
- or `FOR NO KEY UPDATE`
- or `FOR KEY SHARE`

If I am correct, was it not better to introduce a MODE based on ENUM values?
I do not like the possibility setting `ForUpdate` and `ForShare` and `ForNotKeyUpdate` and `ForKeyShare` to `TRUE` or `FALSE` all at the same time (via Java Code).

Also, please format the Grammar properly, its hard to read what is happening here.

Reply:
👍🏻  enumeration is a good way.

Reply:
You can check `java/net/sf/jsqlparser/statement/select/KSQLWindow.java` for the Enum `WindowType` to get a useful template.